kernel/vmvalues.ml

```diff
@@ -27,9 +27,122 @@ let _ = init_vm ()
 
 (* Values of the abstract machine *)
 type values
+type structured_values = values
 let val_of_obj v = ((Obj.obj v):values)
 let crazy_val = (val_of_obj (Obj.repr 0))
 
+type tag = int
+
+let accu_tag = 0
+
+let type_atom_tag = 2
+let max_atom_tag = 2
+let proj_tag = 3
+let fix_app_tag = 4
+let switch_tag = 5
+let cofix_tag = 6
+let cofix_evaluated_tag = 7
+
+(* It would be great if OCaml exported this value,
+   So fixme if this happens in a new version of OCaml *)
+let last_variant_tag = 245
```

Comment:
Fortunately, this is called `Obj.last_non_constructor_tag`. Unfortunately, it was introduced in OCaml 4.02.2 while Coq is meant to compile with 4.02.1. So nothing we can do right now except changing the comment maybe. By the way, it would also be worth asserting that `Obj.first_non_constructor_tag` is 0. (Again, nothing we can implement right now.)

Reply:
Yes, this is on my list. As soon as we updated the required version of OCaml, I'll improve the code.

Reply:
Wait a minute! We do require 4.02.3, in fact. So I'll improve the code, but as part of a separate PR, since this was already there, it is just moved by this PR.

Reply:
We do require 4.02.3.

Reply:
No we don't.

```
if caml_version_nums >= [4;2;1] then
  cprintf "You have OCaml %s. Good!" caml_version
else
  ...
  die "You need OCaml 4.02.1 or later."
```

Reply:
That's a bug in configure.